Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2023 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import Mathlib.Analysis.Convex.Cone.Dual
+import Mathlib.Algebra.Order.Nonneg.Module
+import Mathlib.Algebra.Module.Submodule.Basic
+
+/-!
+# Pointed cones
+
+A *pointed cone* is defined to be a submodule of a module where the scalars are restricted to be
+nonnegative. This is equivalent to saying that as a set a pointed cone is convex cone which
+contains `0`. This is a bundled version of `ConvexCone.Pointed`. We choose the submodule definition
+as it allows us to use the `Module` API to work with convex cones.
+
+## TODO
+
+- Rewrite proper cones using pointed cones.
+
+-/
+
+variable {𝕜 E F G : Type*}
+
+-- TODO: remove `prettyPrint := false` once #6833 is merged
+local notation3 (prettyPrint := false) "𝕜≥0" => {c : 𝕜 // 0 ≤ c}
+
+/-- A pointed cone is a submodule of a module with scalars restricted to being nonnegative. -/
+abbrev PointedCone (𝕜 E) [OrderedSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] :=
+  Submodule {c : 𝕜 // 0 ≤ c} E
+
+namespace PointedCone
+
+open Function
+
+section Definitions
+
+variable [OrderedSemiring 𝕜]
+variable [AddCommMonoid E] [Module 𝕜 E]
+
+instance : Coe (PointedCone 𝕜 E) (ConvexCone 𝕜 E) where
+  coe := fun S => {
+    carrier := S
+    smul_mem' := fun c hc _ hx => S.smul_mem ⟨c, le_of_lt hc⟩ hx
+    add_mem' := fun _ hx _ hy => S.add_mem hx hy }
+
+theorem coe_injective : Injective ((↑) : PointedCone 𝕜 E → ConvexCone 𝕜 E) :=
+  fun _ _ => by simp
+
+@[simp]
+theorem coe_pointed (S : PointedCone 𝕜 E) : (S : ConvexCone 𝕜 E).Pointed := by
+  simp [ConvexCone.Pointed]
+
+@[ext]
+theorem ext {S T : PointedCone 𝕜 E} (h : ∀ x, x ∈ S ↔ x ∈ T) : S = T :=
+  SetLike.ext h
+
+instance instZero (S : PointedCone 𝕜 E) : Zero S :=
+  ⟨0, S.zero_mem⟩
+
+/-- The `PointedCone` constructed from a pointed `ConvexCone`. -/
+def _root_.ConvexCone.toPointedCone {S : ConvexCone 𝕜 E} (hS : S.Pointed) : PointedCone 𝕜 E where
+  carrier := S
+  add_mem' := fun hx hy => S.add_mem hx hy
```

Comment:
```suggestion
  add_mem' hx hy := S.add_mem hx hy
```

Reply:
```suggestion
  add_mem' := S.add_mem
```
maybe even?

Reply:
The second one doesn't work. Lean is having trouble figuring out the implicit variables.